index.bs

```diff
@@ -1610,7 +1914,253 @@ level APIs. The list above includes interface classes for which such high level
+    1. Let |descriptors| as a result of <a>to find a list of descriptors for a configuration</a>
+        with |configurationValue|.
+    1. Let |endpointDescriptors| as an empty <a>list</a>.
+    1. Let |index| be 0.
+    1. While |index| is smaller than the size of |descriptors|:
+        1. Let |descriptor| be |descriptors|[|index|]
+        1. If <code>bDescriptorType</code> of |descriptor| is not equal to <code>INTERFACE</code>,
+            advance |index| by 1 and continue.
+        1. if <code>bInterfaceNumber</code> of |descriptor| is not equal |interfaceNumber|,
+            advance |index| by 1 and continue.
+        1. if <code>bAlternateSetting</code> of |descriptor| is not equal |alternateSetting|,
+            advance |index| by 1 and continue.
+        1. If <code>bNumEndpoints</code> is equal to 0, return |endpointDescriptors|.
+        1. Let |numEndpoints| be <code>bNumEndpoints</code> of |descriptor|.
+        1. Let |indexEndpoints| be 0.
+        1. Let |offset| be |index| + 1
+        1. While |indexEndpoints| is smaller than |numEndpoints|:
+            1. <a>Append</a> |descriptors|[|indexEndpoints| + |offset|] to |endpointDescriptors|.
+            1. Advance |indexEndpoints| by 1.
+        1. Return |endpointDescriptors|.
+</div>
+
+### Constructors ### {#usbalternateinterface-constructors}
+
+<dl dfn-type=constructor dfn-for="USBAlternateInterface">
+    : <dfn>USBAlternateInterface(deviceInterface,alternateSetting)</dfn>
+    ::
+        <div algorithm="USBAlternateInterface constructor">
+            1. Let |deviceInterface| be the parameter of <code>deviceInterface</code>.
+            1. Let |alternateSetting| be the parameter of <code>alternateSetting</code>.
+            1. Set {{USBAlternateInterface/[[interface]]}} to |deviceInterface|.
+            1. Set {{USBAlternateInterface/[[alternateSetting]]}} to be |alternateSetting|.
+            1. Let |descriptors| be the result of <a>to find a list of endpoint descriptors</a> with 
+                <code>interfaceNumber</code> set to |deviceInterface|.{{USBInterface/interfaceNumber}},
+                <code>alternateSetting</code> set to |alternateSetting|,
+                and <code>configurationValue</code> set to
+                |deviceInterface|.{{USBInterface/[[configuration]]}}.{{USBConfiguration/[[configurationValue]]}}.
+            1. <a>For each</a> |descriptor| of |descriptors|:
+                1. If <code>bmAttributes</code> of |descriptor| indicates it is a 
+                    Control Transfer Type (i.e. bits <code>0..1</code> is <code>00</code> 
+                    according to <a>endpoint descriptor</a>), continue.
+                1. Let |endpointAddress| be <code>bEndpointAddress</code> of |descriptor|.
+                1. Let |dir| be <code>null</code>.
+                1. If <code>|endpointAddress| & 0x80</code> is <code>0</code>, set |dir| to {{USBDirection/"out"}}.
+                1. Otherwise, set |dir| to {{USBDirection/"in"}}.
```

Comment:
```suggestion
                1. Let |dir| be {{USBDirection/"out"}} if
                   <code>|endpointAddress| & 0x80</code> is <code>0</code>,
                   and {{USBDirection/"in"}} otherwise.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/1a96f493fc285da99d2b812d809c241388197beb.